include/seqan3/alignment/matrix/detail/trace_iterator_base.hpp

```diff
@@ -68,12 +68,8 @@ class trace_iterator_base
 
     //!\brief Befriend with corresponding const_iterator.
     template <typename other_derived_t, two_dimensional_matrix_iterator other_matrix_iter_t>
-    //!\cond
-#if !SEQAN3_WORKAROUND_FURTHER_CONSTRAIN_FRIEND_DECLARATION
```

Comment:
you are removing the usage of `SEQAN3_WORKAROUND_FURTHER_CONSTRAIN_FRIEND_DECLARATION` without removing the workaround itself?